src/ObjectPool/src/DefaultObjectPool.cs

```diff
@@ -66,6 +67,7 @@ public override T Get()
     /// <inheritdoc />
     public override void Return(T obj)
     {
+        ArgumentNullThrowHelper.ThrowIfNull(obj);
```

Comment:
Can `null` just be a nop instead of throwing?

For diagnostics fail early may seem better, perf-wise maybe just ignore `null` may be faster?!

Reply:
If app code is passing null here and is worried about the overhead of the exception being thrown, then I think the app code can have a null check on their end before calling the API.

Reply:
do u note that NET7 does not behave this way?
i prefer nop anyway, otherwise i would seek to implement a custom objectpoolpolicy to sovle this. why not just keep it simple?

Reply:
I generally agree with the argument that passing null to `Return` indicates a bug in the caller (you didn't get null from the pool, so why would you be returning it?) and that alerting callers to such bugs is generally helpful.  On the other hand, the 7.0 implementation does appear to have gracefully accepted null and just ignored it in subsequent `Get` calls.  I think I'd want to see evidence that the evidence could actually prevent real problems (i.e. in consuming programs) before I'd accept a breaking change.

TL;DR: It's a close call but I'd lean towards noop, rather than throw.

Reply:
> I generally agree with the argument that passing null to `Return` indicates a bug in the caller (you didn't get null from the pool, so why would you be returning it?) and that alerting callers to such bugs is generally helpful. 

The life cycle of a pooled object should not necessarily be a closed loop, some new instances might be fed to return() or some references to old instances might also evolve to be null with the instance disposed, as to decouple the Get/Return lifecycle management.  Coupling'em is a loose choice rather than a tight must.

otherwise, as a practical tool, objectpool should be designed to work as smoothy as possible while considering performance/gracefullness the sametime, instead of asking for unncessary care from programmers from time to time, for it is still not quite efficient/convenient to use.
For complicated cases, such as with multi-references or non-synchronizable life-cycles of referenced underlying children objects, it's a big challege to manange/organize objectpools properly, as not to return a still referenced/already disposed(i.e. null) object, since no way to determine referencec count (which is the fundamental problem for using objectpool), this is the biggest uncertainty.  
So maxmize the torlerance of uncertainty of objectpool is a good principle, by swallowing null checks silently is just one small dent, in my point of view.